### PR TITLE
feat: updated SQL script

### PR DESCRIPTION
**Pull request by IvanFarfan08**
UML Diagram matching the updated SQL script:
image
Sample users, teams and orgs:
image

### CODE REVIEW

File: initdb/init.sql [hunk 1]
```diff
@@ -1,100 +1,65 @@
...
-    name varchar(255) not null,
-    foreign key (team_id) references teams(id),
-    project_proposal_url varchar(255) not null,
-    date_of_recruitment timestamp not null
-)
+    ALTER TABLE Team
+    ADD CONSTRAINT fk_teamLeadId FOREIGN KEY (teamLeadId) REFERENCES Member(memberId),
+    ADD CONSTRAINT fk_projectManagerId FOREIGN KEY (projectManagerId) REFERENCES Member(memberId);
 
-create table blogs  (
```
Comment (miguel-merlin) [inline initdb/init.sql:48]: Don't delete the blogs

File: initdb/init.sql [hunk 2]
```diff
@@ -1,100 +1,65 @@
--- Users
-create table roles (
-	id bigint primary key generated always as identity,
-	name varchar(50) not null check (name in ('E-BOARD', 'TEAM_LEAD', 'PRODUCT_MANAGER', 'DEVELOPER', 'BLUEPRINT_INTERNAL_TEAM'))
-);
+    -- Enum type for roles
```
Comment (miguel-merlin) [inline initdb/init.sql:1]: You have to keep the same conventions in the SQL script. All tables, pk, and fk should be camel case.

File: initdb/init.sql [hunk 3]
```diff
@@ -1,100 +1,65 @@
...
+    CREATE TYPE member_role AS ENUM ('E-BOARD', 'TEAM_LEAD', 'PRODUCT_MANAGER', 'DEVELOPER', 'BLUEPRINT_INTERNAL_TEAM');
 
--- Roles Insert
-insert into roles (name)
-values ('E-BOARD'), ('TEAM_LEAD'), ('PRODUCT_MANAGER'), ('DEVELOPER'), ('BLUEPRINT_INTERNAL_TEAM');
+    -- Create organization table
+    CREATE TABLE Organization (
```
Comment (miguel-merlin) [inline initdb/init.sql:5]: `organizations` not `Organization`

File: initdb/init.sql [hunk 4]
```diff
@@ -1,100 +1,65 @@
...
+        orgId INTEGER GENERATED ALWAYS AS IDENTITY NOT NULL,
```
Comment (miguel-merlin) [inline initdb/init.sql:6]: Why orgId? The ORM will complain about this. Use `id`

File: initdb/init.sql [hunk 5]
```diff
@@ -1,100 +1,65 @@
...
+        orgName VARCHAR(255) NOT NULL,
```
Comment (miguel-merlin) [inline initdb/init.sql:7]: Why are you abbreviating the columns? Also, why are you calling it orgName? If the attribute `name` is in the `organizations` table, it is implicit that the attribute `name` belongs to the `organization`

File: initdb/init.sql [hunk 6]
```diff
@@ -1,100 +1,65 @@
...
-	username varchar(255) not null,
-	email varchar(255) not null,
-	password varchar(255) not null,
-	has_blueprint_email varchar(255) not null,
-	is_enabled boolean not null,
-	date_joined timestamp not null,
-	team_id bigint
-);
+    -- Create Team table
+    CREATE TABLE Team (
```
Comment (miguel-merlin) [inline initdb/init.sql:12]: Same comment here about naming conventions.

File: initdb/init.sql [hunk 7]
```diff
@@ -1,100 +1,65 @@
...
+        teamId INTEGER GENERATED ALWAYS AS IDENTITY,
```
Comment (miguel-merlin) [inline initdb/init.sql:13]: `id` not `teamId`

File: initdb/init.sql [hunk 8]
```diff
@@ -1,100 +1,65 @@
...
+        orgId INTEGER NOT NULL,
```
Comment (miguel-merlin) [inline initdb/init.sql:14]: Don't abbreviate please

File: initdb/init.sql [hunk 9]
```diff
@@ -1,100 +1,65 @@
--- Users
-create table roles (
-	id bigint primary key generated always as identity,
-	name varchar(50) not null check (name in ('E-BOARD', 'TEAM_LEAD', 'PRODUCT_MANAGER', 'DEVELOPER', 'BLUEPRINT_INTERNAL_TEAM'))
-);
+    -- Enum type for roles
+    CREATE TYPE member_role AS ENUM ('E-BOARD', 'TEAM_LEAD', 'PRODUCT_MANAGER', 'DEVELOPER', 'BLUEPRINT_INTERNAL_TEAM');
```
Comment (miguel-merlin) [inline initdb/init.sql:2]: Change `BLUEPRINT_INTERNAL_TEAM` to `TECH_TEAM`

File: initdb/init.sql (hunk 3 above)
Comment (miguel-merlin) [inline initdb/init.sql:5]: Organization is missing:
- Product Manger
- Team Leads

File: initdb/init.sql (hunk 6 above)
Comment (miguel-merlin) [inline initdb/init.sql:12]: Add a designer field as well

File: initdb/init.sql [hunk 10]
```diff
@@ -1,100 +1,65 @@
...
-);
+    -- Create Member table
+    CREATE TABLE Member (
+        memberId INTEGER GENERATED ALWAYS AS IDENTITY,
+        teamId INTEGER,
+        name VARCHAR(255) NOT NULL,
+        username VARCHAR(255) NOT NULL,
+        email VARCHAR(255) UNIQUE,
+        password VARCHAR(255) NOT NULL,
+        role member_role,
```
Comment (miguel-merlin) [inline initdb/init.sql:31]: This should be a set of roles. i.e. a member can have multiple roles (e-board, team lead, etc).

File: initdb/init.sql [hunk 11]
```diff
@@ -1,100 +1,65 @@
...
+    ADD CONSTRAINT fk_teamLeadId FOREIGN KEY (teamLeadId) REFERENCES Member(memberId),
+    ADD CONSTRAINT fk_projectManagerId FOREIGN KEY (projectManagerId) REFERENCES Member(memberId);
 
-create table blogs  (
-    id bigint primary key generated always as identity,
-    author varchar(255) not null,
-    title varchar(255) not null,
-    date_created timestamp not null
-)
+    CREATE INDEX idx_member_teamid ON Member(teamId);
```
Comment (miguel-merlin) [inline initdb/init.sql:44]: What is the purpose of this?

**Review by miguel-merlin — CHANGES_REQUESTED** [review]
Pleas keep the naming conventions to `snake_case`. The name of the tables should be plural `Organization`->`organizations`. Do not abbreviate the attributes. If the organization has a name, do not set the attribute to `orgName`, this is redundant. Use bigint for ids, not `INTEGER`

**Comment by miguel-merlin** [comment]
@IvanFarfan08 Good first iteration, please resolve the comments I made.

**Comment by IvanFarfan08** [comment]
image

File: initdb/init.sql [hunk 12]
```diff
@@ -1,100 +1,108 @@
...
-
--- Insert blogs to the blogs table
-insert into blogs (author, title, date_created) values
-('John Doe', 'Blog 1', CURRENT_TIMESTAMP),
-('Jane Smith', 'Blog 2', CURRENT_TIMESTAMP),
-('Michael Johnson', 'Blog 3', CURRENT_TIMESTAMP);
+    -- Update Team table to set team_lead_id, project_manager_id and designer_id after members have been inserted
+    UPDATE teams SET team_lead_id = 2, project_manager_id = 2, designer_id = 3 WHERE id = 1;
+    UPDATE teams SET team_lead_id = 1, project_manager_id = 1, designer_id = 3 WHERE id = 2;
+    UPDATE organizations SET team_lead_id = 2, project_manager_id = 2 WHERE id = 1;
```
Comment (miguel-merlin) [inline initdb/init.sql]: **nit**: Add space here for convention

File: initdb/init.sql [hunk 13]
```diff
@@ -1,100 +1,108 @@
...
+   -- Insert sample teams
+   INSERT INTO teams (organization_id, name, team_lead_id, project_manager_id, designer_id, date_created) VALUES
+   (1, 'Frontend Team', NULL, NULL, NULL, '2021-09-01 00:00:00'),
+   (2, 'Backend Team', NULL, NULL, NULL, '2024-10-01 00:00:00');
 
+    -- Insert sample members
+    INSERT INTO members (team_id, name, username, email, password, is_active) VALUES
+    (1, 'John Doe', 'jdoe', 'jdoe@stevens.edu', 'password', TRUE),
+    (2, 'Jane Smith', 'janesmith', 'janesmith@stevens.edu', 'password2', TRUE),
+    (2, 'Alice Johnson', 'alicej', 'alicej@stevens.edu', 'a1234', FALSE);
```
Comment (miguel-merlin) [inline initdb/init.sql]: Remove

**Review by miguel-merlin — APPROVED** [review]

**Pull request merged**